File: r/raindrops/raindrops.R
raindrops <- function(n) {
    df <- data.frame(
           d = c(3, 5, 7),
        word = c("Pling", "Plang", "Plong")
    )

    res <- paste(df[n %% df$d == 0,]$word, collapse="")

    if (res == "") as.character(n) else res
}

File: ruby/raindrops/raindrops.rb
class Raindrops
  def self.convert(n)
    result = { Pling: 3, Plang: 5, Plong: 7}
      .map { |w,d| n % d == 0 ? w.to_s : "" }
      .join("")

    result == "" ? n.to_s : result
  end
end
